libraries/freertos_plus/standard/freertos_plus_tcp/source/FreeRTOS_IP.c

```diff
@@ -2104,248 +2120,251 @@ uint16_t ucVersionHeaderLength;
 
 BaseType_t location = 0;
 
-	/* Check for minimum packet size. */
-	if( uxBufferLength < sizeof( IPPacket_t ) )
-	{
-	 	usChecksum = ipINVALID_LENGTH;
-		location = 1;
-	 	goto error_exit;
-	}
-
-	/* Parse the packet length. */
-	pxIPPacket = ipPOINTER_CAST( const IPPacket_t *, pucEthernetBuffer );
-
-	/* Per https://tools.ietf.org/html/rfc791, the four-bit Internet Header
-	Length field contains the length of the internet header in 32-bit words. */
-	ucVersionHeaderLength = pxIPPacket->xIPHeader.ucVersionHeaderLength;
-	ucVersionHeaderLength = ( ucVersionHeaderLength & ( uint8_t ) 0x0FU ) << 2;
-	uxIPHeaderLength = ( UBaseType_t ) ucVersionHeaderLength;
-
-	/* Check for minimum packet size. */
-	if( uxBufferLength < ( sizeof( IPPacket_t ) + ( uxIPHeaderLength - ipSIZE_OF_IPv4_HEADER ) ) )
-	{
-	 	usChecksum = ipINVALID_LENGTH;
-		location = 2;
-	 	goto error_exit;
-	}
-	usLength = pxIPPacket->xIPHeader.usLength;
-	usLength = FreeRTOS_ntohs( usLength );
-	if( uxBufferLength < ( size_t ) ( ipSIZE_OF_ETH_HEADER + ( size_t ) usLength ) )
+	/* Introduce a do-while loop to allow use of break statements.
+	 * Note: MISRA prohibits use of 'goto', thus replaced with breaks. */
+	do
```

Comment:
It's interesting that we chose ```do {...} while( ipFALSE_BOOL );``` instead of something like ``` if ( continue == pdTrue ) {...}```. Both could work if done properly, the latter looks more "linear"...

Though with do-while loop, are we missing a break at the very end of the iteration? So that if nothing goes wrong with the first iteration, this function exits? Maybe I missed?

Reply:
It is `do{ ... } while ( 0 );` essentially. So no need of a `break`.